webapp/src/api/controller/BookController.java

```diff
@@ -136,19 +149,39 @@ public static String getCallbackPath() {
 
 
     @RequestMapping(value = "/api/changeBookName",
-            method = RequestMethod.PUT)
-    public HashMap<String, Object> changeBookName(@RequestHeader("auth-token") String authToken,
-                                                  @RequestBody String json) {
+            method = {RequestMethod.PUT, RequestMethod.GET})
+    //public HashMap<String, Object> changeBookName(@RequestHeader("auth-token") String authToken,
+                                                  //@RequestBody String json) {
+    public HashMap<String, Object> changeBookName(@RequestBody String json) {
+        System.out.println(json);
         JSONObject obj = new JSONObject(json);
+        System.out.println(obj);
         String bookId = (String) obj.get("bookId");
-        if (!Authorization.authorizeBook(bookId, authToken)){
-            JsonWrapper.generateError("Permission denied for accessing this book");
-        }
+        System.out.println(bookId);
+        //if (!Authorization.authorizeBook(bookId, authToken)){
+            //JsonWrapper.generateError("Permission denied for accessing this book");
+        //}
         String newBookName = (String) obj.get("newBookName");
+        String query = "SELECT COUNT(*) FROM books WHERE bookname = ?";
```

Comment:
Change the query to
`SELECT EXISTS( SELECT FROM books WHERE bookname = ? )`
The output is a boolean.

The advantage of `EXISTS` is (from PostgreSQL documentation)
> The subquery will generally only be executed long enough to determine whether at least one row is returned, not all the way to completion.